packages/repository/src/relations/has-many/has-many.repository.ts

```diff
@@ -114,4 +114,12 @@ export class DefaultHasManyRepository<
       options,
     );
   }
+
+  async count(where?: Where<TargetEntity>, options?: Options): Promise<Count> {
```

Comment:
@achrinza great work!, I believe this is needed especially for lb3 to lb4 migration. I believe you should add the count method here also in the interface `HasManyRepository`  contract above.

Reply:
> You should add the count method also in the interface HasManyRepository to keep its contract inSync with the implementation.

Thanks for the reminder, @marioestradarosa! I'll make the amendments. This is a draft PR as I'd like to implement all the necessary interfaces requested by #5025 (and more, if any).